src/Address/IPv4.php

```diff
@@ -41,6 +41,30 @@ class IPv4 implements AddressInterface
      */
     protected $comparableString;
 
+    /**
+     * The list of IPv4 RFC designated address ranges.
+     *
+     * @var mixed
+     */
+    public static $reservedRanges = array(
```

Comment:
What about making this private and caching its values?
I mean, something like this:
```php
/*
 * @var array|null
 */
private static $reservedRanges = null;
/*
 * @return array
 */
public static getReservedRanges()
{
    if (self::$reservedRanges === null) {
        $reservedRanges = array();
        foreach(array(
            '0.0.0.0/32' => RangeType::T_UNSPECIFIED),      //RFC 5735
            '0.0.0.0/8' => RangeType::T_THISNETWORK),      //RFC 5735
            // ...
        ) as $range => $type) {
            $reservedRanges[] = array('range' => Subnet::fromString($range), 'type' => $type);
        }
        self::$reservedRanges = $reservedRanges;
    }
    return self::$reservedRanges;
}
```

Reply:
Seems sensible to me. :)